setup-fmdeps.sh

```diff
@@ -162,6 +162,10 @@ else
   echo "Creating opam switch ${OPAM_SWITCH_NAME}."
   opam switch create --empty --repositories="${OPAM_SELECTED_REPOS}" \
     "${OPAM_SWITCH_NAME}"
+  # `opam env --switch=A --set-switch` followed by `opam env --switch=B` leaves us writing to switch `A` instead of switch `B`.
+  # Ask `opam env` to revert its side-effects on the environment, then clear OPAMSWITCH (which --revert ignores).
+  eval $(opam env --revert)
+  unset -v OPAMSWITCH OPAM_SWITCH_PREFIX OPAM_LAST_ENV
   # Avoid --set-switch here, it would hide misconfigurations from the $(opam switch show) test
   eval $(opam env --switch="${OPAM_SWITCH_NAME}")
```

Comment:
What about
```suggestion
  # Clear any user-chosen OPAMSWITCH, so that we use the new switch instead
  unset OPAMSWITCH
  # Avoid --set-switch here, it would hide misconfigurations from the $(opam switch show) test
  eval $(opam env)
```

All other effects of `opam env` done by the user should be overridden by the next `opam env`, no?

Reply:
https://github.com/ocaml/opam/issues/5823 suggests this is more complicated in edge cases — I'm not  `unset OPAM_SWITCH_PREFIX OPAM_LAST_ENV` still looks like messing with opam internals, tho `opam` behavior might be questionable.

Also, on my machine `opam env --revert` clears `OPAM_SWITCH_PREFIX` as expected.

Reply:
I (test and) accept your suggestion. (I have a note somewhere that opam env isn't idempotent which is why I started with the revert but the only relevant state here should be OPAMSWITCH.)